src/super_gradients/training/models/segmentation_models/segformer.py

```diff
@@ -0,0 +1,599 @@
+    the bounds. The method used for generating the random values works
+    best when :math:`a \leq \text{mean} \leq b`.
+    Args:
+        tensor: an n-dimensional `torch.Tensor`
+        mean: the mean of the normal distribution
+        std: the standard deviation of the normal distribution
+        a: the minimum cutoff value
+        b: the maximum cutoff value
+    Examples:
+        >>> w = torch.empty(3, 5)
+        >>> nn.init.trunc_normal_(w)
+    """
+    return _no_grad_trunc_normal_(tensor, mean, std, a, b)
+
+
+class PatchEmbedding(nn.Module):
+    def __init__(self, in_channels: int, out_channels: int, patch_size: int, stride: int, padding: int):
+        """
+        Overlapped patch merging (https://arxiv.org/pdf/2105.15203.pdf)
+        :param in_channels: number of input channels
+        :param out_channels: number of output channels (embedding dimension)
+        :param patch_size: patch size (k for size (k, k))
+        :param stride: patch stride (k for size (k, k))
+        :param padding:  patch padding (k for size (k, k))
+        """
+
+        super().__init__()
+
+        self.proj = nn.Conv2d(in_channels=in_channels, out_channels=out_channels, kernel_size=patch_size, stride=stride, padding=padding)
+        self.norm = nn.LayerNorm(out_channels)
+
+    def forward(self, x: torch.Tensor) -> tuple[torch.Tensor, int, int]:
+        x = self.proj(x)
+        _, _, h, w = x.shape
+
+        x = x.flatten(2).transpose(1, 2)
+        x = self.norm(x)
+
+        return x, h, w
+
+
+class EfficientSelfAttention(nn.Module):
+    def __init__(self, dim: int, head: int, sr_ratio: int):
+        """
+        Efficient self-attention (https://arxiv.org/pdf/2105.15203.pdf)
+        :param dim: embedding dimension
+        :param head: number of attention heads
+        :param sr_ratio: the reduction ratio of the efficient self-attention
+        """
+
+        super().__init__()
+
+        self.head = head
+        self.sr_ratio = sr_ratio
+        self.scale = (dim // head) ** -0.5
+        self.q = nn.Linear(dim, dim)
+        self.kv = nn.Linear(dim, dim * 2)
+        self.proj = nn.Linear(dim, dim)
+
+        if sr_ratio > 1:
+            self.sr = nn.Conv2d(dim, dim, sr_ratio, sr_ratio)
+            self.norm = nn.LayerNorm(dim)
+
+    def forward(self, x: torch.Tensor, h: int, w: int) -> torch.Tensor:
+        b, n, c = x.shape
+        q = self.q(x).reshape(b, n, self.head, c // self.head).permute(0, 2, 1, 3)
+
+        if self.sr_ratio > 1:
+            x = x.permute(0, 2, 1).reshape(b, c, h, w)
+            x = self.sr(x).reshape(b, c, -1).permute(0, 2, 1)
+            x = self.norm(x)
+
+        k, v = self.kv(x).reshape(b, -1, 2, self.head, c // self.head).permute(2, 0, 3, 1, 4)
+
+        attn = (q @ k.transpose(-2, -1)) * self.scale
+        attn = attn.softmax(dim=-1)
+
+        x = (attn @ v).transpose(1, 2).reshape(b, n, c)
+        x = self.proj(x)
+        return x
+
+
+class DropPath(nn.Module):
```

Comment:
Possible duplicate of super_gradients.training.utils.regularization_utils.DropPath

Reply:
Changed to SG implementation